packages/edit-post/src/store/selectors.js

```diff
@@ -104,34 +150,49 @@ const MIGRATED_KEYS = [
  *
  * @return {Object} Preferences Object.
  */
-export const getPreferences = createRegistrySelector(
-	( select ) => ( state ) => {
-		const editPostPreferences = state.preferences;
-
-		// Some preferences now exist in the preferences store.
-		// Fetch them so that they can be merged into the post
-		// editor preferences.
-		const preferenceStorePreferences = MIGRATED_KEYS.reduce(
-			( accumulatedPrefs, preferenceKey ) => {
-				const value = select( preferencesStore ).get(
-					'core/edit-post',
-					preferenceKey
-				);
-
-				return {
-					...accumulatedPrefs,
-					[ preferenceKey ]: value,
-				};
-			},
-			{}
+export const getPreferences = createRegistrySelector( ( select ) => () => {
+	deprecated( `wp.data.select( 'core/edit-post' ).getPreferences`, {
+		since: '6.0',
+		alternative: `wp.data.select( 'core/preferences' ).get`,
+	} );
+
+	// These preferences now exist in the preferences store.
+	// Fetch them so that they can be merged into the post
+	// editor preferences.
+	const preferences = [
+		'hiddenBlockTypes',
+		'editorMode',
+		'preferredStyleVariations',
+	].reduce( ( accumulatedPrefs, preferenceKey ) => {
+		const value = select( preferencesStore ).get(
+			'core/edit-post',
+			preferenceKey
 		);
 
 		return {
-			...editPostPreferences,
-			...preferenceStorePreferences,
+			...accumulatedPrefs,
+			[ preferenceKey ]: value,
 		};
-	}
-);
+	}, {} );
+
+	// Panels were a preference, but the data structure changed when the state
+	// was migrated to the preferences store. They need to be converted from
+	// the old to new format.
```

Comment:
This is a little confusing as it says "They need to be converted from the old to new format", but they are passed through a method called ToOld format?

Reply:
What is the context that causes the need to have them converted back to the old format, I am having trouble trying to work out what combination of editor version versus stored data would need that - apologies if I am overlooking the obvious.

Reply:
Ah, that looks like a typo, thanks for catching it. I'll fix that 😄 

> What is the context that causes the need to have them converted back to the old format?

It's purely if a plugin is using the `getPreference` and/or `getPreferences` selector for panel data, it would be a breaking change to suddenly provide that data in the new format. So while it's painful, I think the best thing to do is convert to the old format. The good thing is these selectors are now deprecated, so plugins should be moving away from using them.

Reply:
That makes sense - I completely overlooked plugins when trying to think this through!